### PR TITLE
Lib199 Drivetrain Changes

### PR DESCRIPTION
**Pull request by CoolSpy3**
This branch has basically become the defacto branch for lib199 changes this season (because they're all for drivetrain and drivetrain requires all of them) and has expanded wayyy past its original scope. At this point (despite all changes not being fully tested on the robot), my guess is that the public API will remain fairly stable, and I think it's probably best to just merge this into master and add any necessary fixes as we need them.

Notice: This PR contains breaking changes to the pathing interfaces in order to streamline their implementation.

### CODE REVIEW

File: src/main/java/org/carlmontrobotics/lib199/path/SwerveDriveInterface.java [hunk 1]
```diff
@@ -117,25 +100,34 @@ public default Command createAutoCommand(Trajectory trajectory, Supplier<Rotatio
...
+     * @param eventMap   Map of event marker names to the commands that should run when reaching that marker.
+     *                   This SHOULD NOT contain any commands requiring Drivetrain, or it will be interrupted
+     * @return PPSwerveControllerCommand
+     */
+    public default Command createPPAutoCommand(List<PathPlannerTrajectory> trajectory, HashMap<String, Command> eventMap) {
         Subsystem[] requirements = eventMap.values().stream().flatMap(command -> command.getRequirements().stream())
                 .toArray(Subsystem[]::new);
         requirements = Arrays.copyOf(requirements, requirements.length + 1);
         requirements[requirements.length - 1] = this;
         requirements = Arrays.stream(requirements).distinct().toArray(Subsystem[]::new);
```
Comment (brettle) [inline src/main/java/org/carlmontrobotics/lib199/path/SwerveDriveInterface.java:120]: I don't see where `requirements` is used.

**Review by brettle — APPROVED** [review]
Recommend addressing my one comment if possible, but it isn't critical.

File: src/main/java/org/carlmontrobotics/lib199/path/SwerveDriveInterface.java (hunk 1 above)
Comment (CoolSpy3) [reply src/main/java/org/carlmontrobotics/lib199/path/SwerveDriveInterface.java:120]: My bad. I rewrote that command a few times and totally forgot to use that correctly. It's been fixed.

**Comment by CoolSpy3** [comment]
After I merged this, it looked like the javadoc build failed on master, so I've pushed one more commit outside of this PR that fixes those and updates the GitHub Action so that, in the future, javadoc build (but not deploy) is run on all branches (not just master).

**Pull request merged**